src/common/options.d

```diff
@@ -0,0 +1,10 @@
+module common.options;
+
+/// Common Options struct
+struct Options
+{
+    bool assume_yes;
+}
+
+/// Globally accessible options
+public Options options;
```

Comment:
This actually looks too specific to be useful in `common`, it's normal (and expected) for apps to evolve to have different options and it's not saving much typing and there is no much room for bugs to be introduced in here.

Reply:
Some functions in other common modules reference the global options, that's why it's here.

Reply:
Sounds like about time to refactor all involved code.

Reply:
Well.. originally it was different.... then luca came a long.. https://github.com/sociomantic-tsunami/neptune/pull/95#discussion_r139154186

Reply:
You can do both by extracting into dedicated package utility that takes configuration via argument and then inside every application package defining small wrappers that get those arguments from app config.

Reply:
Oh, OK, I get it now, you are defining a global options module because some function use a global from that module? In that case OK, it is ugly indeed but practical. I wouldn't mind if you prefer to go back to passing the `bool` explicitly via arguments now that you are abstracting out some functions, you could just define a new function passing that `bool` from a global option if you want to save the extra typing everywhere.

Reply:
Maybe I won't call it options, as it seems too general, or even maybe I would consider to move that to the affected module instead. Like in `shellHelper` you could just define a global `public` bool `assume_yes` and that's it.

Reply:
But anyway, I see different trade-offs in the different schemes and I'm not particularly crazy about any of them, so feel free to make it as it would make more sense to you.

Reply:
I guess I could do that as well (defining `assume_yes` in `shellHelper`).. maybe a bit nicer than the current solution

Reply:
I want to stress again my original point again - if you are splitting something as a common utility, you should spend time cleaning technical debt and redesigning it as independent self-contained library package. We are not at early prototype stage where "getting something to work" is the main priority, one extra day to clean up stuff is totally warranted. And there is not place for globals in library configuration.

Reply:
> if you are splitting something as a common utility, you should spend time cleaning technical debt and redesigning it as independent self-contained library package

And I have to say I don't agree with this, I think is a nice thing to do but not something **required**. This is not supposed to be a generally useful library in itself, is just a toolbox or shared code for the different commands here, so it might be *hacky* if that is useful for this particular project. You don't need to think about potential new users and have many abstractions for shared code inside a project.

Reply:
Are you supposed to think about potential reviewers? Reasonable excuses to introduce hacky code knowingly could be some sort of time pressure or lack of information about required abstractions. In this case we have neither. With attitude like that, next time I will need to change anything in neptune, it will be faster to rewrite it from scratch.

Reply:
I think there is an intermediate, good code is still good code even if is not abstract enough to be used everywhere. When reviewing code in a repo you need to know a bit of context in that repo. I'm not talking about extremes. Just to stick to this example, for me having a global variable in the `shellHelper` module `assume_yes` to configure the behaviour of that module is acceptable, while in a general purpose library for me that won't be acceptable. This is the kind of things I'm talking about.